File: Leetcode Challenge/February/java/Summary Ranges.java
public List<String> summaryRanges(int[] nums) {
    int length = nums.length;
    List<String> result = new ArrayList<String>(length);
    for (int i = 0; i < length; i++) {
        int num = nums[i];
        while (i < length - 1 && nums[i] + 1 == nums[i + 1]) {
            i++;
        }
        if (num != nums[i]) {
            result.add(num + "->" + nums[i]);
        } else {
            result.add(num + "");
        }
    }
    return result;
}

File: Leetcode Challenge/February/cpp/Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        const int size_n = nums.size();
        vector<string> res;
        if ( 0 == size_n) return res;
        for (int i = 0; i < size_n;) {
            int start = i, end = i;
            while (end + 1 < size_n && nums[end+1] == nums[end] + 1) end++;
            if (end > start) res.push_back(to_string(nums[start]) + "->" + to_string(nums[end]));
            else res.push_back(to_string(nums[start]));
            i = end+1;
        }
        return res;
    }
};
